### ISSUE TITLE
[RETEST] Mobile A11Y: Filter tags vocalized as links and don't notify users clicking on them will clear the filter

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13767]
**Additional context**

- Fix this ticket only after the whole Screen Reader milestone for desktop is completed. 

**Describe the bug**

- Results Page: E.g. price range tag filter is vocalized as a link and doesn't notify the user that clicking on it will **clear** the filter.

**To Reproduce: Watch Test Videos**

- Android Talkback test video
https://video.sap.com/media/t/1_1gtfpn6o
min 0:25:19
  
- Iphone VoiceOver test video
https://video.sap.com/media/t/1_qa70rxtl
min 0:09:17
_Different behavior: not vocalized, perhaps tester didn't allow enough time, needs re-testing._

**Expected behavior**
-  Should be read as a button (action happens on the same page) and also extend vocalization to _"click to reset range"_ so the user knows what will happen on click.
